pkg/apis/awsnodetemplate/v1alpha1/awsnodetemplate.go

```diff
@@ -29,6 +31,17 @@ type AWSNodeTemplateSpec struct {
 	// +optional
 	UserData     *string `json:"userData,omitempty"`
 	v1alpha1.AWS `json:",inline"`
+	// AMIs is a list of custom AMIs that Karpenter will choose from. If multiple AMIs satisfy your pod requirements,
+	// one will be chosen at random.
+	// +optional
+	AMIs []AMI `json:"amis,omitempty"`
+}
+
+type AMI struct {
+	// Id is an identifier for an AMI.
+	ID string `json:"id,omitempty"`
+	// Properties defines all the properties of an AMI like architecture, GPU compatibility and so on.
+	Properties map[string]string `json:"properties,omitempty"`
```

Comment:
See design guidelines. Reuse k8s concepts where applicable.
```suggestion
	Selector map[string]string `json:"properties,omitempty"`
```

Reply:
I thought about this and the word `Selector` didn't make sense to me. `nodeSelector` makes sense in a podSpec -> you're selecting a set of nodes that the pod can be assigned to. 

This is not helping you select an AMI since the AMI is already defined by this struct. You're only defining the requirements of an AMI to be used / or it's properties.

Reply:
Selector (LabelSelector, NodeSelector, etc) is a broadly used concept in the k8s that is identical to the semantic you're defining here. I could see an argument for `AMISelector`, similar to `SubnetSelector`, but given that this is already a property of AMI, to avoid stuttering you'd need `ami.selector`

Reply:
I still don't agree. `SubnetSelector` helps you choose / select a subnet based on something. This isn't helping you select an AMI, but is rather just defining what the AMI represents. Same for label selector / node selector. 

I don't think it's the same semantic.

Reply:
You're right, it's the inverse semantic. The pattern is typically `Selector` -> `Labels`. The thing selecting (e.g. the pod) uses a selector, where the thing targeted has labels. Perhaps we can simply call this `labels`?

Reply:
As is, I would agree w/ @suket22 that this is not the same concept and would be confusing to conflate the two. It could be modified to be the same and might be useful at the expense of describing the AMIs and discovering the attributes on them.

Reply:
Not sure I'm tracking @bwagner5. Perhaps we can discuss offline?

Reply:
Adding amiSelector in a subsequent commit.